### PR TITLE
Feature/dapp reusable action forms and progress

### LINKED ISSUE #2890: dApp - make open channel and deposit to channel re-usable
## Description

Needed to be used within #2882
The logic of the actions is currently fixed in the routes which are not re-usable. We need to get it out there into a plain component. The action step handling in the dApp is a nightmare and might become reworked here.

## Acceptance criteria
- 

## Tasks
- [ ]

### PR DESCRIPTION
**Pull request by weilbith**
Fixes #2890
Based on #2920

**Short description**
TBD

**Definition of Done**

- [ ] Steps to manually test the change have been documented
- [ ] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1.
2.

### CODE REVIEW

File: raiden-dapp/src/components/channels/ChannelActionForm.vue [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    this.partnerAddressInput = partnerAddress;
+  }
+
+  @Watch('tokenAmount', { immediate: true })
+  onTokenAmountChange(tokenAmount: string): void {
+    this.tokenAmountInput = tokenAmount;
+  }
+
+  @Ref()
+  readonly actionReference!: typeof ActionMixin;
```
Comment (taleldayekh) [inline raiden-dapp/src/components/channels/ChannelActionForm.vue:191]: Can be removed.

File: raiden-dapp/src/components/channels/ChannelActionForm.vue [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+  @Emit('completed')
+  emitCompleted(): void {
+    // pass
+  }
+
+  @Emit('failed')
+  emitFailed(error: Error): Error {
+    return error;
+  }
```
Comment (taleldayekh) [inline raiden-dapp/src/components/channels/ChannelActionForm.vue:250]: Can be removed.

File: raiden-dapp/src/components/channels/ChannelDepositDialog.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <raiden-dialog data-cy="channel-deposit-dialog" visible @close="emitClose">
+    <v-card-title>{{ $t('channel-deposit.title') }}</v-card-title>
+
+    <v-card-text>
+      <channel-deposit-action
+        #default="{ runAction, confirmButtonLabel }"
+        :title="$t('channel-deposit.title')"
```
Comment (taleldayekh) [inline raiden-dapp/src/components/channels/ChannelDepositDialog.vue:8]: Can possibly be removed.

File: raiden-dapp/src/components/channels/ChannelActionForm.vue (hunk 1 above)
Comment (weilbith) [reply raiden-dapp/src/components/channels/ChannelActionForm.vue]: ✔️

File: raiden-dapp/src/components/channels/ChannelActionForm.vue (hunk 2 above)
Comment (weilbith) [reply raiden-dapp/src/components/channels/ChannelActionForm.vue]: ✔️

File: raiden-dapp/src/components/channels/ChannelDepositDialog.vue (hunk 3 above)
Comment (weilbith) [reply raiden-dapp/src/components/channels/ChannelDepositDialog.vue:8]: ✔️

**Comment by weilbith** [comment]
Alright. Talel and I did a "pair-programming-review" together yesterday. The review comments above are in fact mine as I found out that the linter did not detect those unused code snippets (part of older implementation approach versions).
Talel did agree with the general new concept of the actions I have introduced. The actual review and approval is outstanding. He might wanna check for example the new translations.
Today he is traveling for work and can't do it. But as he approved the general approach, I can continue with my work for #2882 using this PR.

File: raiden-dapp/tests/e2e/utils/blockchain-interaction.ts [hunk 4]
```diff
@@ -33,11 +33,9 @@ export function mintConnectedTokenFromSelectHubScreen() {
 export function openChannel() {
   cy.log('open channel');
   // cypress selectors: raiden-dapp/src/views/OpenChannelRoute.vue
```
Comment (taleldayekh) [inline raiden-dapp/tests/e2e/utils/blockchain-interaction.ts:35]: Maybe delete this comment as well.

**Review by taleldayekh — APPROVED** [review]
Your walkthrough the other day gave me a good understanding of the changes. I skimmed through it now and could not find anything apparent that seemed out of order.

Some nit-picks about spelling but no big deal, but you should address the ones in the locales the others you can decide yourself.

Thank you! 👍🏼

File: raiden-dapp/src/locales/en.json [hunk 5]
```diff
@@ -263,6 +263,7 @@
       "empty": "Enter an address or ENS name",
       "ens-resolve-failed": "Please enter a resolvable ENS name",
       "invalid-excluded-address": "Please enter a different address or ENS name. You can't use your own address or the address of the token network.",
+      "invalid-restricted-address": "Please enter a different address or ENS name. The used address is not allowed to be used here.",
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/src/locales/en.json:266]: Gets resolved [here](https://github.com/raiden-network/light-client/pull/2920#discussion_r701894253).

File: raiden-dapp/src/components/channels/ChannelDepositAction.vue [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  title: 'transfer.steps.deposit.title',
+  description: 'transfer.steps.deposit.description',
+  active: false,
+  completed: false,
+  failed: false,
+};
+
+@Component({ components: { ActionProgressCard, RaidenDialog } })
+export default class ChannelDepositAction extends Mixins(ActionMixin) {
+  depositStep = Object.assign({}, cleanDepositStep);
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/src/components/channels/ChannelDepositAction.vue:18]: It is basically a simple technique to make a deep clone of an object. There are three different one or so...

File: raiden-dapp/tests/e2e/utils/blockchain-interaction.ts (hunk 4 above)
Comment (weilbith) [reply raiden-dapp/tests/e2e/utils/blockchain-interaction.ts:35]: We have tons of these comments all over the place. I think deleting just this one as well won't be helpful. We might wanna open an issue to finally remove all of them. 😬

**Comment by weilbith** [comment]
I now first rebased this on `master` after #2920 got merged and resolved the conflicts. I'll now push fixup commits to address the review feedback so it is easier to see that I solved them.

File: raiden-dapp/tests/unit/components/address-input.spec.ts [hunk 7]
```diff
@@ -243,6 +248,35 @@ describe('AddressInput', () => {
     });
   });
 
+  describe('restricted addresses', () => {
+    test('should show error message when the user enters address not part of the restrited list', async () => {
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/components/address-input.spec.ts]: ✔️

File: raiden-dapp/tests/unit/components/amount-input.spec.ts [hunk 8]
```diff
@@ -151,4 +155,33 @@ describe('AmountInput.vue', () => {
       expect(wrapper.vm.$data.amount).toBe('');
     });
   });
+
+  describe('adapt zero input values to changing decimals', () => {
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/components/amount-input.spec.ts]: ✔️

File: raiden-dapp/src/locales/en.json [hunk 9]
```diff
@@ -293,6 +293,9 @@
     "settle": "Settle"
   },
   "channel-deposit": {
+    "title": "Channel Deposit",
+    "restricted-partner-error": "You can only uses addresses you have an open channel with.",
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/src/locales/en.json [hunk 10]
```diff
@@ -570,8 +571,11 @@
       "deposit-failed": "Could not deposit to the channel.",
       "open-failed": "Channel open failed."
     },
+    "title": "Open Channel",
     "open-button": "Open Channel",
     "partner": "Channel Partner",
+    "deposit": "Initial Deposit",
+    "excluded-partner-error": "You already have an open channel with this address for this token.",
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/src/locales/en.json]: ✔️

File: raiden-dapp/tests/unit/components/channels/channel-action-form.spec.ts [hunk 11]
```diff
@@ -0,0 +1,365 @@
...
+async function submitForm(wrapper: Wrapper<ChannelActionForm>): Promise<void> {
+  const form = wrapper.get('.channel-action-form');
+
+  form.vm.$emit('submit', { preventDefault: () => null });
+  await wrapper.vm.$nextTick();
+}
+
+describe('ChannelActionForm.vue', () => {
+  describe('token address', () => {
+    test('hides form section if set so', async () => {
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/components/channels/channel-action-form.spec.ts]: ✔️

File: raiden-dapp/tests/unit/components/channels/channel-action-form.spec.ts [hunk 12]
```diff
@@ -0,0 +1,365 @@
...
+
+      expect(fetchAndUpdateTokenData).toHaveBeenCalledTimes(2);
+      expect(fetchAndUpdateTokenData).toHaveBeenLastCalledWith(['0xOtherToken']);
+      expect(monitorToken).toHaveBeenCalledTimes(2);
+      expect(monitorToken).toHaveBeenLastCalledWith('0xOtherToken');
+    });
+  });
+
+  describe('partner address', () => {
+    test('hides form section if set so', async () => {
```
[the remark this answers is not in the document]
Comment (weilbith) [reply raiden-dapp/tests/unit/components/channels/channel-action-form.spec.ts]: ✔️

**Comment by weilbith** [comment]
Alright. I addressed all review comments. As they were all about spelling, I'm rushing a little and squash the fixup commits already.

**Comment by weilbith** [comment]
Ah, some feedback here was actually about changes in the predecessor PR. Thereby the fixup commits do not work. I need to combine them into a new standalone commit.

**Comment by weilbith** [comment]
I suffer from the TypeScript update...

**Comment by weilbith** [comment]
Alright. TypeScript issues fixed and CI is fine. So I squash the fixup commits. No CHANGELOG entry required.

**Pull request merged**